src/jsonpatcherproxy.js

```diff
@@ -122,46 +163,50 @@ var JSONPatcherProxy = (function() {
                 value: receiver
               });
             }
-            return Reflect.set(target, key, receiver);
+            return ownReflectSet(instance, target, key, receiver);
           } else {
             instance.defaultCallback({
               op: 'replace',
               path: distPath,
               value: receiver
             });
-            return Reflect.set(target, key, receiver);
+            return ownReflectSet(instance, target, key, receiver);
           }
         } else {
           instance.defaultCallback({
             op: 'add',
             path: distPath,
             value: receiver
           });
-          return Reflect.set(target, key, receiver);
+          return ownReflectSet(instance, target, key, receiver);
         }
       },
       deleteProperty: function(target, key) {
         if (typeof target[key] !== 'undefined') {
           instance.defaultCallback({
             op: 'remove',
-            path: path +
-              '/' +
-              JSONPatcherProxy.escapePathComponent(key.toString())
+            path:
+              path + '/' + JSONPatcherProxy.escapePathComponent(key.toString())
           });
-          const proxyInstance = proxifiedObjectsMap.get(target[key]);
+          const proxyInstance = instance.proxifiedObjectsMap.get(target[key]);
+
           if (proxyInstance) {
-            instance.disableTrapsForProxy(proxyInstance);
+            instance.objectsPathsMap.delete(proxyInstance.originalObject);
+            instance.disableTrapsForProxy(proxyInstance.proxy);
+            instance.proxifiedObjectsMap.delete(target[key]);
```

Comment:
Have you considered `WeakMap` instead of `Map`? Then you would not need to `delete` items from it.